### PR TITLE
Decompile `no0` Skelerang entities

### PR DESCRIPTION
**Pull request by JoshSchreuder**
~~Planning on giving this a tidy up when I import data and do some analysis in debugger so I'll fix up some of the naming then unless there's something glaring that sticks out here.~~

`func_us_801D191C` / `EntitySkelerangUnknown` was previous decompiled but slightly updated here for PSP match:
https://decomp.me/scratch/8lEH0

`func_us_801D20A4` / `EntitySkelerangBoomerang` matching both
PSX: https://decomp.me/scratch/SMgwb
PSP: https://decomp.me/scratch/WFD0w

`func_us_801D191C` / `EntitySkelerang` matching both
PSX: https://decomp.me/scratch/BUkq2
PSP: https://decomp.me/scratch/OYg4D

Thanks to Discord gang for helping with some PSP stuff here ^

### CODE REVIEW

File: src/st/no0/e_skelerang.c [hunk 1]
```diff
@@ -2,17 +2,335 @@
...
+                MakeEntityFromId(E_SKELERANG, self, entity);
+                entity->flags = FLAG_DESTROY_IF_OUT_OF_CAMERA |
+                                FLAG_POS_CAMERA_LOCKED | FLAG_UNK_2000;
+                entity->hitboxState = 0;
+                entity->animCurFrame = i + 44;
+                entity->params = i;
+                entity->facingLeft = self->facingLeft;
+                if ((GetSideToPlayer() & 1) ^ 1) {
+                    // Sus store to Entity 0xA
+                    F(entity->velocityX).i.hi = -(Random() & 3);
```
Comment (JoshSchreuder) [inline src/st/no0/e_skelerang.c:183]: A suggestion from @bismurphy to pull this into an f32 macro - I guess it's possible `velocityX` / `velocityY` are actually f32s and accessed this way everywhere but there's no other evidence of that in the codebase besides here so I've just macroed for now.

File: src/st/no0/e_skelerang.c [hunk 2]
```diff
@@ -2,17 +2,351 @@
...
+        if (entity != NULL) {
+            CreateEntityFromEntity(2, self, entity);
+            entity->params = 1;
+        }
+        DestroyEntity(self);
+        break;
+    }
+}
+
+void EntitySkelerangUnknown(Entity* entity) {
```
Comment (JoshSchreuder) [inline src/st/no0/e_skelerang.c:343]: Even in debugger I wasn't really able to ascertain what this is. It gets initialised as you approach the skelerangs but other than that I can't tell what it's doing. I can't trigger the destroy function in gameplay so this is a bit of a mystery

**Review by sozud — APPROVED** [review]

**Comment by sozud** [comment]
Seems to have picked up issues after updating to master
```

mipsel-linux-gnu-ld: build/us/src/st/no0/e_init.c.o: in function `NO0_EntityUpdates':
src/st/no0/e_init.c:(.data+0xb4): undefined reference to `func_us_801D191C'
mipsel-linux-gnu-ld: src/st/no0/e_init.c:(.data+0xb8): undefined reference to `func_us_801D20A4'
mipsel-linux-gnu-ld: src/st/no0/e_init.c:(.data+0xbc): undefined reference to `func_us_801D2318'
mipsel-linux-gnu-ld: build/us/src/st/no0/e_skelerang.c.o: in function `LM19':
src/st/no0/e_skelerang.c:(.text+0xec): undefined reference to `g_EInitSkelerang'
mipsel-linux-gnu-ld: src/st/no0/e_skelerang.c:(.text+0xf0): undefined reference to `g_EInitSkelerang'
mipsel-linux-gnu-ld: build/us/src/st/no0/e_skelerang.c.o: in function `LM172':
src/st/no0/e_skelerang.c:(.text+0x7c8): undefined reference to `g_EInitSkelerangBoomerang'
mipsel-linux-gnu-ld: src/st/no0/e_skelerang.c:(.text+0x7cc): undefined reference to `g_EInitSkelerangBoomerang'
mipsel-linux-gnu-ld -nostdlib --no-check-sections -o build/us/stsel.elf -Map build/us/stsel.map -T build/us/stsel.ld -T config/
```

**Comment by JoshSchreuder** [comment]
Needed to update e_init data since #1898, have done now so should be good

**Pull request merged**